File: Python/core/Matrix.py
#!/usr/bin/env python3 

from collections import deque

class Matrix:
   """Matrix representation of various components of the program"""
   def __init__ ( self ):
      self._width = 1
      self._height = 1
      self._id = 0
      self._contents = []
      
   def print ( self ):
      for i in range ( 0 , self._height ):
         print ( self._contents[ i ] )
         
   def load ( self , file__contents ):
      self._width = int ( file__contents.popleft () )
      self._height = int ( file__contents.popleft () )
      for i in range ( 0 , self._height ):
         row = []
         for j in range ( 0 , self._width ):
            tmp = int ( file__contents.popleft () )
            row.append ( tmp )
            if tmp > 0:
               self._id = tmp
         self._contents.append ( row )
   
   def getWidth ( self ):
      return self._width
   
   def getHeight ( self ):
      return self._height
   
   def getPoint ( self , x , y ):
      return self._contents[ y ][ x ]
   
   def getId ( self ):
      return self._id

File: CPP/src/core/matrix/matrix.cc
# include "matrix.hh"

Core::Matrix::Matrix ( void )
{
   width = 1;
   height = 1;
   id = 0;
}

Core::Matrix::~Matrix ( void )
{
}
      
void Core::Matrix::print ( void )
{
   for ( unsigned int i = 0; i < contents.size (); i++ )
   {
      for ( unsigned int j = 0; j < contents[ i ].size (); j++ )
      {
         cout << contents[ i ][ j ] << " ";
      }
      cout << endl;
   }
   
   return;
}
      
void Core::Matrix::load ( ifstream& file )
{
   file >> width >> height;
   
   for ( int i = 0; i < height; i++ )
   {
      vector< int > row;
      for ( int j = 0; j < width; j++ )
      {
         int tmp;
         file >> tmp;
         row.push_back ( tmp );
         
         if ( tmp > 0 )
         {
            id = tmp;
         }
      }
      contents.push_back ( row );
   }
}
      
int Core::Matrix::getWidth ( void )
{
   return ( width );
}
      
int Core::Matrix::getHeight ( void )
{
   return ( height );
}
      
int Core::Matrix::getPoint ( int x , int y )
{
   return ( contents[ y ][ x ] );
}
      
int Core::Matrix::getId ( void )
{
   return ( id );
}
